lib/doorkeeper/config.rb

```diff
@@ -70,8 +70,30 @@ def scopes_by_grant_type(hash = {})
       # `params` object.
       #
       # @param methods [Array] Define client credentials
+      # @deprecated
       def client_credentials(*methods)
```

Comment:
This is pretty messy because of all the different ways `client_credentials` could be set, and we can only convert known methods across automatically.

We could move the deprecated line to the `else` at the end, and basically say: "here's what you need instead: client_authentication [...]"